src/main/java/net/dv8tion/jda/core/entities/impl/JDAImpl.java

```diff
@@ -467,7 +535,7 @@ public void shutdown()
     }
 
     @Override
-    public JDAClient asClient()
+    public JDAClientImpl asClient()
```

Comment:
Why?

Reply:
I need the implementation of JDABot to use the ShardManager setter and thought if I change that method this should be changed too. There's also no reason to hide the real class from implementation only methods as it will just lead to more casts.